packages/lit-html/src/directives/unsafe-html.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {
+  directive,
+  Directive,
+  nothing,
+  TemplateResult,
+  noChange,
+  PartInfo,
+  NODE_PART,
+} from '../lit-html.js';
+
+const HTML_RESULT = 1;
+
+export class UnsafeHTML extends Directive {
+  static directiveName = 'unsafeHTML';
+  static resultType = HTML_RESULT;
+
+  value: unknown = nothing;
+  templateResult?: TemplateResult;
+
+  constructor(part: PartInfo) {
+    super();
+    if (part.type !== NODE_PART) {
+      throw new Error(
+        `${
+          (this.constructor as typeof UnsafeHTML).directiveName
+        }() can only be used in text bindings`
+      );
+    }
+  }
+
+  render(value: string) {
+    // TODO: add tests for nothing and noChange
+    if (value === nothing) {
+      this.templateResult = undefined;
+      return (this.value = value);
+    }
+    if (value === noChange) {
+      return value;
+    }
+    if (typeof value != 'string') {
+      throw new Error(
+        `${
+          (this.constructor as typeof UnsafeHTML).directiveName
+        }() called with a non-string value`
+      );
+    }
+    if (value === this.value) {
+      return this.templateResult;
+    }
+    this.value = value;
+    const strings = ([String(value)] as unknown) as TemplateStringsArray;
```

Comment:
We're coercing the value to a String already; this seems fine, so I'd just remove the value type check and error. I guess do one or the other...

Reply:
done